lib/system.nim

```diff
@@ -347,6 +347,15 @@ proc arrPut[I: Ordinal;T,S](a: T; i: I;
 proc `=destroy`*[T](x: var T) {.inline, magic: "Destroy".} =
   ## Generic `destructor`:idx: implementation that can be overridden.
   discard
+
+proc `=dup`[T](x: var ref T, y: ref T) {.inline, magic: "Dup".} =
```

Comment:
No.

Reply:
It is a private helper function, which can be used to ease code generation.

Reply:
Name it differently then.